src/coreComponents/constitutive/solid/SolidModelDiscretizationOps.hpp

```diff
@@ -35,6 +35,23 @@ namespace constitutive
  */
 struct SolidModelDiscretizationOps
 {
+  /**
+   * @brief Compute inner product matrix for solid mechanics
+   * @tparam NUM_SUPPORT POINTS Number of support points (nodes) for this element
+   * @tparam BASIS_GRADIENT Finite element shape function gradients type
+   * @tparam CBF Callback function
+   * @param gradN Finite Element shape function gradients
+   * @param elementStiffness Local stiffness matrix
+   * @param callbackFunction The callback function
+   */
+  template< int NUM_SUPPORT_POINTS,
+            typename BASIS_GRADIENT,
+            typename CBF >
+  GEOSX_HOST_DEVICE
+  void BTDB( BASIS_GRADIENT const & gradN,
+             real64 ( &elementStiffness )[NUM_SUPPORT_POINTS*3][NUM_SUPPORT_POINTS*3],
+             CBF && callbackFunction );
```

Comment:
Maybe give a more meaningful name to the lambda function to make sure that we know what it is supposed to do, and the variables it is supposed to modify.

Reply:
I just followed what was done in `upperBTDB`, `diagBTDB`, `diagRowSumBTDB`. However, I agree we should use better naming convention. I suggest we address this point in a dedicated PR.